sql/updates/world/master/9999_99_99_99_world_heavens_wrath.sql

```diff
@@ -0,0 +1,7 @@
+DELETE FROM `spell_script_names` WHERE `spell_id` = 421558;
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
+(421558, 'spell_pri_heavens_wrath');
+
+DELETE FROM `spell_proc` WHERE `SpellId` IN (421558);
+INSERT INTO `spell_proc` (`SpellId`,`SchoolMask`,`SpellFamilyName`,`SpellFamilyMask0`,`SpellFamilyMask1`,`SpellFamilyMask2`,`SpellFamilyMask3`,`ProcFlags`,`ProcFlags2`,`SpellTypeMask`,`SpellPhaseMask`,`HitMask`,`AttributesMask`,`DisableEffectsMask`,`ProcsPerMinute`,`Chance`,`Cooldown`,`Charges`) VALUES
+(421558,0x02,6,0x00000000,0x00818000,0x00000080,0x00000000,0x0,0x0,0x3,0x2,0x403,0x0,0x0,0,0,0,0); -- Heaven's Wrath
```

Comment:
SchoolMask is optional (setting it to holy is also wrong because that causes this talent to not work with Power of the Dark Side - it does on retail)

SpellFamilyMask is wrong, you made it too broad
should be only 0x00018000 in `SpellFamilyMask1`, 0 in other fields

It also shouldn't proc from Ultimate Penitence bolts (they have the same SpellFamilyMask as penance bolts) - you will need to exclude that in script

Reply:
any tips on how to exclude the bolts of that spell?

Reply:
No other option other than hardcode spell ids

Reply:
im not sure what you mean by this

Reply:
> SpellFamilyMask is wrong, you made it too broad should be only 0x00018000 in `SpellFamilyMask1`, 0 in other fields

but i found Penance in the other masks too, so why wouldn't it be fine?

Reply:
check spellid in CheckProc

Reply:
> > SpellFamilyMask is wrong, you made it too broad should be only 0x00018000 in `SpellFamilyMask1`, 0 in other fields
> 
> but i found Penance in the other masks too, so why wouldn't it be fine?

You want to find a minimal mask that covers all the spells you want (and nothing else)
And that is these four: 373130, 400187, 47666 and 47750

Reply:
> > > SpellFamilyMask is wrong, you made it too broad should be only 0x00018000 in `SpellFamilyMask1`, 0 in other fields
> > 
> > 
> > but i found Penance in the other masks too, so why wouldn't it be fine?
> 
> You want to find a minimal mask that covers all the spells you want (and nothing else) And that is these four: 373130, 400187, 47666 and 47750

aight
so as i thought